src/hotspot/share/gc/z/zLock.inline.hpp

```diff
@@ -84,6 +85,65 @@ inline bool ZReentrantLock::is_owned() const {
   return owner == thread;
 }
 
+inline ZConditionLock::ZConditionLock() {
+  pthread_cond_init(&_cond, NULL);
+  pthread_mutex_init(&_mutex, NULL);
+}
+
+inline ZConditionLock::~ZConditionLock() {
+  pthread_cond_destroy(&_cond);
+  pthread_mutex_destroy(&_mutex);
+}
+
+inline void ZConditionLock::lock() {
+  pthread_mutex_lock(&_mutex);
+}
+
+inline bool ZConditionLock::try_lock() {
+  return pthread_mutex_trylock(&_mutex) == 0;
+}
+
+inline void ZConditionLock::unlock() {
+  pthread_mutex_unlock(&_mutex);
+}
+
+inline bool ZConditionLock::wait(uint64_t millis) {
+  assert(millis >= 0, "negative timeout");
+  if (millis > 0) {
+    jlong timeout = millis * (NANOUNITS / MILLIUNITS);
+
+    struct timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
+
+    // Calculate a new absolute time that is "timeout" nanoseconds from "now".
+    struct timespec abstime;
+    jlong seconds = timeout / NANOUNITS;
+    abstime.tv_sec = now.tv_sec + seconds;
+
+    timeout %= NANOUNITS; // remaining nanos
+    long nanos = now.tv_nsec + timeout;
+    if (nanos >= NANOUNITS) { // overflow
+      abstime.tv_sec += 1;
+      nanos -= NANOUNITS;
+    }
+    abstime.tv_nsec = nanos;
+
+    int status = pthread_cond_timedwait(&_cond, &_mutex, &abstime);
+    return status == 0;
+  } else {
+    pthread_cond_wait(&_cond, &_mutex);
```

Comment:
lock the mutex

Reply:
no worries. the `_mutex` is locked in https://github.com/openjdk/jdk/commit/d7e68f375cf063#diff-933933faab3bf6064037e90e26af3c35dab98042a9d5a68c0acb8462ebaf2f50R43, the following patch for uncommitting memory.